spec.html

```diff
@@ -36959,11 +36957,11 @@ <h1>RegExp.prototype [ @@replace ] ( _string_, _replaceValue_ )</h1>
           1. Let _functionalReplace_ be IsCallable(_replaceValue_).
           1. If _functionalReplace_ is *false*, then
             1. Set _replaceValue_ to ? ToString(_replaceValue_).
-          1. Let _global_ be ! ToBoolean(? Get(_rx_, *"global"*)).
+          1. Let _flags_ be ? ToString(? Get(_rx_, *"flags"*)).
+          1. If _flags_ contains *"g"*, let _global_ be *true*. Otherwise, let _global_ be *false*.
           1. If _global_ is *true*, then
-            1. Let _fullUnicode_ be ! ToBoolean(? Get(_rx_, *"unicodeSets"*)).
-            1. If _fullUnicode_ is *false*, set _fullUnicode_ to ! ToBoolean(? Get(_rx_, *"unicode"*)).
             1. Perform ? Set(_rx_, *"lastIndex"*, *+0*<sub>𝔽</sub>, *true*).
+          1. If _flags_ contains *"u"* or _flags_ contains *"v"*, let _fullUnicode_ be *true*. Otherwise, let _fullUnicode_ be *false*.
```

Comment:
Decoupling assignment of _fullUnicode_ to no longer be conditioned on _global_ being true reads so much better!